### PR TITLE
Implement homeserver selection

### PR DESCRIPTION
**Pull request by JorikSchellekens**
This implements the options required for matrix.to to adhere to GDPR privacy requirements. The options let the user chose which homeservers the service can use to fetch invite metadata. 

[Hosted Version](matrix-to-static-4n657dkmd.vercel.app)

### CODE REVIEW

**Comment by jryans** [comment]
@JorikSchellekens: As a random aside, GitHub treats the bare domain for the hosted version as a relative URL, so it doesn't go to the right place. It seems like you need to add `https://` for the right thing to happen.

File: src/components/HomeserverOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                dispatcher({ action: ActionType.SetHS, HSURL: HSUrl })
+            }
+        >
+            <Form>
+                <Input
+                    type="text"
+                    name="HSUrl"
+                    placeholder="https://example.com"
+                />
+                <Button type="submit">Set HS</Button>
```
Comment (jryans) [inline src/components/HomeserverOptions.tsx:72]: Seems strange to display "HS" to the user here and nowhere else?

```suggestion
                <Button type="submit">Set homeserver</Button>
```

File: src/components/HomeserverOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    dispatcher({ action: ActionType.SetAny });
+                }}
+            >
+                Use any homeserver
+            </Button>
+            <Toggle
+                checked={usePrefered}
+                onChange={(): void => setUsePrefered(!usePrefered)}
+            >
+                Use my prefered homeserver only
```
Comment (jryans) [inline src/components/HomeserverOptions.tsx:107]:
```suggestion
                Use my preferred homeserver only
```

File: src/components/HomeserverOptions.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                <img
+                    src={icon}
+                    alt="Icon making it clear that connections may be made with external services"
+                />
+            </div>
+            <StyledCheckbox
+                checked={rememberSelection}
+                onChange={(e): void => setRemeberSelection(e.target.checked)}
+            >
+                Remember my choice.
```
Comment (jryans) [inline src/components/HomeserverOptions.tsx:94]: At least in Element, we typically leave the full stop off of toggles like this.

```suggestion
                Remember my choice
```

**Review by jryans — APPROVED** [review]
Looks great overall, thanks! 😄

File: src/components/HomeserverOptions.tsx (hunk 1 above)
Comment (JorikSchellekens) [reply src/components/HomeserverOptions.tsx:72]: This is changing with design review

**Comment by JorikSchellekens** [comment]
@jryans thanks for the heads up about the links. Sorry for making you fix all those. Vercel auto copies a bare domain on deploy

**Pull request merged**